svc/pkg/workflow/db/workflow/migrations/20240430191643_init.up.sql

```diff
@@ -0,0 +1,73 @@
+CREATE TABLE nodes (
+  id UUID PRIMARY KEY,
+  last_ping_ts INT
+);
+
+-- TODO: In the event of a node failure, clear all of the wake conditions and remove the node id. This can be
+-- done in a periodic GC service
+CREATE TABLE workflows (
+  id UUID PRIMARY KEY,
+  name TEXT NOT NULL,
+  -- The node that's running this workflow
+  node_id UUID,
+  input TEXT NOT NULL,
+  -- Null if incomplete
+  output TEXT,
+
+  wake_immediate BOOLEAN NOT NULL DEFAULT false,
+  wake_deadline_ts INT,
+  wake_signals TEXT[] NOT NULL DEFAULT ARRAY[],
+  wake_sub_workflow_id UUID
+);
+
+CREATE INDEX gin_workflows_wake_signals
+ON workflows
+USING GIN (wake_signals);
+
+-- Stores activity outputs for replay
+CREATE TABLE workflow_activity_events (
+  workflow_id UUID NOT NULL REFERENCES workflows,
+  location INT[] NOT NULL,
+  name TEXT NOT NULL,
+  -- CRDB can't store u64, so we have to store bytes
+  input_hash BYTES NOT NULL,
+  -- Null if incomplete
+  output TEXT,
+
+  PRIMARY KEY (workflow_id, location)
+);
+
+-- Stores acknowledged signals for replay
+CREATE TABLE workflow_signal_events (
+  workflow_id UUID NOT NULL REFERENCES workflows,
+  location INT[] NOT NULL,
+  id TEXT NOT NULL,
```

Comment:
i don't undertsand how this is used in the code. is this a mistake? no idea what this is.

Reply:
Signals need a unique id